src/rviz_tf_publisher.cpp

```diff
@@ -0,0 +1,92 @@
+ *  
+ *  3. Neither the name of the copyright holder nor the names of its contributors may be used to
+ *  endorse or promote products derived from this software without specific prior written permission.
+ *  
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ *  IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ *  FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ *  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ *  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ *  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ *  IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+ *  OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ ****************************************************************************************************/
+
+/*
+ * Author(s) : Andy McEvoy ( mcevoy.andy@gmail.com )
+ * Desc      : implementation of rviz_tf_publisher.h. See *.h file for documentation.
+ * Created   : 2017 - May - 18
+ */
+
+#include <tf_keyboard_cal/rviz_tf_publisher.h>
+
+namespace tf_keyboard_cal
+{
+
+RvizTFPublisher::RvizTFPublisher()
+  : nh_("~")
+{
+  create_tf_sub_ = nh_.subscribe("/rviz_tf_create", 10, &RvizTFPublisher::createTF, this);
+  remove_tf_sub_ = nh_.subscribe("/rviz_tf_remove", 10, &RvizTFPublisher::removeTF, this);
+  update_tf_sub_ = nh_.subscribe("/rviz_tf_update", 10, &RvizTFPublisher::updateTF, this);
+}
+
+void RvizTFPublisher::createTF(geometry_msgs::TransformStamped create_tf_msg)
+{
+  active_tfs_.push_back(create_tf_msg);
+}
+
+void RvizTFPublisher::removeTF(geometry_msgs::TransformStamped remove_tf_msg)
+{
+  for (std::size_t i = 0; i < active_tfs_.size(); i++)
+  {
+    if (remove_tf_msg.child_frame_id.compare(active_tfs_[i].child_frame_id) == 0 &&
+        remove_tf_msg.header.frame_id.compare(active_tfs_[i].header.frame_id) == 0)
+    {
+      active_tfs_.erase(active_tfs_.begin() + i);
+      break;
+    }                                            
+  }
+}
+
+void RvizTFPublisher::updateTF(geometry_msgs::TransformStamped update_tf_msg)
+{
+  ROS_DEBUG_STREAM_NAMED("updateTF","from: " << update_tf_msg.header.frame_id << ", to: " << update_tf_msg.child_frame_id);
+  for (std::size_t i = 0; i < active_tfs_.size(); i++)
+  {
+    if (update_tf_msg.child_frame_id.compare(active_tfs_[i].child_frame_id) == 0 &&
+        update_tf_msg.header.frame_id.compare(active_tfs_[i].header.frame_id) == 0)
+    {
+      ROS_DEBUG_STREAM_NAMED("updateTF","i = " << i);
+      active_tfs_[i].transform = update_tf_msg.transform;
+    }                                            
+  }
+}
+
+void RvizTFPublisher::publishTFs()
+{
+  static tf::TransformBroadcaster br;
```

Comment:
might be left over from an initial attempt to get the functionality I wanted. I'll review and remove if not needed anymore.